Apps/phairlockdigital/airlockdigital_connector.py

```diff
@@ -0,0 +1,1030 @@
+            action_result.add_debug_data({'r_text': r.text})
+            action_result.add_debug_data({'r_headers': r.headers})
+
+        # Process each 'Content-Type' of response separately
+        # Custom command
+        self.save_progress("Making process response")
+        # Process a json response
+        if 'json' in r.headers.get('Content-Type', ''):
+            return self._process_json_response(r, action_result)
+
+        # Process an HTML response, Do this no matter what the api talks.
+        # There is a high chance of a PROXY in between phantom and the rest of
+        # world, in case of errors, PROXY's return HTML, this function parses
+        # the error and adds it to the action_result.
+        if 'html' in r.headers.get('Content-Type', ''):
+            return self._process_html_response(r, action_result)
+
+        # it's not content-type that is to be parsed, handle an empty response
+        if not r.text:
+            return self._process_empty_response(r, action_result)
+
+        # everything else is actually an error at this point
+        message = "Can't process response from server. Status Code: {0} Data from server: {1}".format(
+                r.status_code, r.text.replace('{', '{{').replace('}', '}}'))
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, endpoint, action_result, method="get", **kwargs):
+        # **kwargs can be any additional parameters that requests.request accepts
+
+        config = self.get_config()
+
+        resp_json = None
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)), resp_json)
+
+        # Create a URL to connect to
+        url = self._base_url + endpoint
+
+        try:
+            r = request_func(
+                            url,
+                            # auth=(username, password),  # basic authentication
+                            verify=config.get('verify_server_cert', False),
+                            **kwargs)
+
+        except Exception as e:
+            return RetVal(action_result.set_status( phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(str(e))), resp_json)
+
+        return self._process_response(r, action_result)
+
+    def _handle_test_connectivity(self, param):
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Add config to init variable to get root API Key
+        config = self.get_config()
+
+        # Place api key in its own variable.
+        api_key = config.get('apiKey')
+
+        # Parameter Dictionary to pass to the request
+        header_var = {
+            "X-APIKey": api_key
+        }
+
+        url = '/license/get'
```

Comment:
Might I suggest creating a airlockdigital_consts.py file and put the constants in that file. This will help in case the API changes, we wont need to come in here and edit each instance of the URL manually. 

A good example of this is to check out https://github.com/phantomcyber/phantom-apps/blob/next/Apps/phcrowdstrikeoauth/crowdstrikeoauthapi_consts.py

Reply:
Thank you for the example. I have added a consts file for all routes as seen in the CrowdStrike app. Much nicer to maintain going forward.